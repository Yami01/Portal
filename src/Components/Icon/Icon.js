/**
 *
 * Icon
 * @format
 * @flow
 *
 */

import React, { memo } from 'react'
import type { AbstractComponent, Node } from 'React'
import { View } from 'react-native'

import IconStyles from './styles'
import type { PropsType } from './types'
import Icons from '@/Assets/Icons'

const Icon: AbstractComponent<PropsType> = memo((props: PropsType): Node => {
  const {
    group,
    name,
    extraStyle,
    withBorder,
    height,
    width,
    fillColor,
    isCenter,
    testID,
  } = props

  const icon = Icons[group]?.(name, extraStyle, width, height, fillColor)

  return (
    <View
      {...testID}
      style={[
        withBorder && IconStyles.iconBorder,
        isCenter && IconStyles.center,
      ]}
    >
      {icon}
    </View>
  )
})

Icon.displayName = 'Icon'

export default Icon
