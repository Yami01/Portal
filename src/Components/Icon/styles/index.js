/**
 *
 * @format
 * @flow
 *
 */

import { StyleSheet } from 'react-native'
import type { IconStyleType } from '../types'
import { Colors } from '@/Theme/Variables'

const IconStyles: IconStyleType = StyleSheet.create({
  center: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  iconBorder: {
    position: 'relative',
    borderColor: Colors.black,
    borderRadius: 5,
    borderWidth: 1,
    overflow: 'hidden',
    padding: 10,
  },
})

export default IconStyles
