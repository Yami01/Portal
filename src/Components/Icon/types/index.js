/**
 *
 * @format
 * @flow
 *
 */

import type { ImageStyleProp, ViewStyleProp } from "react-native/Libraries/StyleSheet/StyleSheet"

export type PropsType = {
  group: string,
  name: string,
  height?: number,
  width?: number,
  fillColor?: string,
  isCenter?: boolean,
  extraStyle?: ImageStyleProp,
  withBorder?: boolean,
  accessibilityLabel?: string,
  accessibilityHint?: string,
  testID?: string,
}

export type IconStyleType = {
  center: ViewStyleProp,
  iconBorder: ViewStyleProp,
}
