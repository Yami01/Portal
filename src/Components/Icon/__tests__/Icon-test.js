/**
 *
 * Icon Test
 * @format
 * @flow
 *
 */

import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react-native'
import Icon from '@/Components/Icon'

const testId = 'iconTestId'

describe('Icon', () => {
  test('is icon render properly', () => {
    const tree = renderer
      .create(<Icon testID={testId} group="general" name="arrow_backward" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('is a11y working in test', () => {
    const { queryByTestId } = render(
      <Icon testID={testId} group="general" name="arrow_backward" />,
    )
    const icon = queryByTestId(testId)
    expect(icon).not.toBeNull()
  })

  test('is icon render properly withBorder', () => {
    const { queryByTestId } = render(
      <Icon testID={testId} group="general" name="arrow_backward" withBorder />,
    )
    const icon = queryByTestId(testId)
    expect(icon).not.toBeNull()
  })

  test('is icon render properly withBorder and center', () => {
    const { queryByTestId } = render(
      <Icon
        testID={testId}
        group="general"
        name="arrow_backward"
        withBorder
        isCenter
      />,
    )
    const icon = queryByTestId(testId)
    expect(icon).not.toBeNull()
  })
})
