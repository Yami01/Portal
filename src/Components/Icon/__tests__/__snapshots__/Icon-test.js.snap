// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon is icon render properly 1`] = `
<View
  0="i"
  1="c"
  2="o"
  3="n"
  4="T"
  5="e"
  6="s"
  7="t"
  8="I"
  9="d"
  style={
    Array [
      undefined,
      undefined,
    ]
  }
/>
`;
